src/server/api/routers/activity/activity-create.procedure.ts

```diff
@@ -0,0 +1,58 @@
+import { and, eq } from "drizzle-orm"
+import shortUuid from "short-uuid"
+
+import { CreateActivityInputDto } from "@/contracts/activity/activity"
+import {
+  sendActivityArchivedEvent,
+  sendActivityCreatedEvent,
+} from "@/contracts/events/activity"
+import { db } from "@/database"
+import { activities } from "@/database/schemas"
+import waitForPredicate from "@/lib/wait-for-predicate"
+import { protectedProcedure } from "@/server/api/trpc"
+
+export const createActivityProcedure = protectedProcedure
+  .input(CreateActivityInputDto)
+  .mutation(async ({ input }) => {
+    // TODO: make sure user has correct permissions
```

Comment:
I'll create a issue for this :)

Reply:
never mind, I pushed a quick solution to this, which was to use our existing admin middleware